### PR TITLE
Fix #27618: Add commands for cleaning up

### LINKED ISSUE #27618: [enterprise-4.6] Issue in file installing/installing_bare_metal_ipi/ipi-install-troubleshooting.adoc
### Which section(s) is the issue in?

In **Cleaning up previous installations**.

### What needs fixing?

The command `sudo virsh vol-delete $i --pool default` did nothing becaulse the volume is not in `default` pool.

```
[kni@provisioner ~]$ for i in $(sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'});
> do
>   sudo virsh destroy $i;
>   sudo virsh undefine $i;
>   sudo virsh vol-delete $i --pool default;
>   sudo virsh vol-delete $i.ign --pool default;
> done
Domain ocp-8pbbl-bootstrap destroyed

Domain ocp-8pbbl-bootstrap has been undefined

error: failed to get vol 'ocp-8pbbl-bootstrap'
error: Storage volume not found: no storage vol with matching path 'ocp-8pbbl-bootstrap'

Vol ocp-8pbbl-bootstrap.ign deleted
```

Therefore, the pool and volume are not deleted correctly after the cleaning up procedure has been invoked:

```
[kni@provisioner ~]$ sudo virsh pool-list --all
 Name                  State    Autostart
-------------------------------------------
 default               active   yes
 ocp-8pbbl-bootstrap   active   yes

[kni@provisioner ~]$ sudo virsh vol-list ocp-ft6ng-bootstrap
 Name                  Path
--------------------------------------------------------------------------------------------------
 ocp-ft6ng-bootstrap   /var/lib/libvirt/openshift-images/ocp-ft6ng-bootstrap/ocp-ft6ng-bootstrap
```

### Workaround

Modify the commands in `for` loop to delete volume and pool correctly.

```
for i in $(sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'});
do
  sudo virsh destroy $i;
  sudo virsh undefine $i;
  sudo virsh vol-delete $i --pool $i;
  sudo virsh vol-delete $i.ign --pool default;
  sudo virsh pool-destroy $i;
  sudo virsh pool-undefine $i;
done
```

### PR DESCRIPTION
**Pull request by kurokobo**
Change the pool name which the volume exists and add commands to delete pools.

At least in **4.6**, this fix seems to be necessary. I have no idea if it is necessary in 4.5 or 4.7.

### CODE REVIEW

**Comment by kurokobo** [comment]
@vikram-redhat Please review this PR.

**Comment by vikram-redhat** [comment]
@johnwilkins can you ack?

@ahardin-rh for peer review afterwards.

**Comment by rlopez133** [comment]
LGTM

File: modules/ipi-install-configuring-the-install-config-file.adoc [hunk 1]
```diff
@@ -97,7 +97,9 @@ for i in $(sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'});
 do
   sudo virsh destroy $i;
   sudo virsh undefine $i;
-  sudo virsh vol-delete $i --pool default;
+  sudo virsh vol-delete $i --pool $i;
```
Comment (rlopez133) [inline modules/ipi-install-configuring-the-install-config-file.adoc:100]:
```
sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'}
```

Only captures the name of the VM with bootstrap in it. $i would never be the pool name. The pool name (by default) is called default.

File: modules/ipi-install-troubleshooting-cleaning-up-previous-installations.adoc [hunk 2]
```diff
@@ -23,8 +23,10 @@ for i in $(sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'});
 do
   sudo virsh destroy $i;
   sudo virsh undefine $i;
-  sudo virsh vol-delete $i --pool default;
+  sudo virsh vol-delete $i --pool $i;
```
Comment (rlopez133) [inline modules/ipi-install-troubleshooting-cleaning-up-previous-installations.adoc:26]: same comment as above

File: modules/ipi-install-configuring-the-install-config-file.adoc (hunk 1 above)
Comment (kurokobo) [reply modules/ipi-install-configuring-the-install-config-file.adoc:100]: Thank you for review. When I tested with 4.6 in my environment, the volume `$i.ign` is in `default` pool but the volume `$i` is in the pool named `$i`.
In 4.5 and earlier both `$i.ign` and `$i` are in `default` pool. It seems the placement of volumes is changed in 4.6? 

This is my log:

```
[kni@provisioner ~]$ sudo virsh list
 Id   Name                  State
-------------------------------------
 1    ocp-dr8v2-bootstrap   running

[kni@provisioner ~]$ sudo virsh pool-list
 Name                  State    Autostart
-------------------------------------------
 default               active   yes
 ocp-dr8v2-bootstrap   active   yes

[kni@provisioner ~]$ sudo virsh vol-list default
 Name                      Path
----------------------------------------------------------------------------
 ocp-dr8v2-bootstrap.ign   /var/lib/libvirt/images/ocp-dr8v2-bootstrap.ign

[kni@provisioner ~]$ sudo virsh vol-list ocp-dr8v2-bootstrap
 Name                  Path
-------------------------------------------------------------------------------------------------
 ocp-dr8v2-bootstrap   /var/lib/libvirt/openshift-images/ocp-dr8v2-bootstrap/ocp-dr8v2-bootstrap
```

There fore the commands in this PR are required for me to clean up everything in 4.6.

```
[kni@provisioner ~]$ for i in $(sudo virsh list | tail -n +3 | grep bootstrap | awk {'print $2'});
> do
>   sudo virsh destroy $i;
>   sudo virsh undefine $i;
>   sudo virsh vol-delete $i --pool $i;
>   sudo virsh vol-delete $i.ign --pool default;
>   sudo virsh pool-destroy $i;
>   sudo virsh pool-undefine $i;
> done
Domain ocp-dr8v2-bootstrap destroyed

Domain ocp-dr8v2-bootstrap has been undefined

Vol ocp-dr8v2-bootstrap deleted

Vol ocp-dr8v2-bootstrap.ign deleted

Pool ocp-dr8v2-bootstrap destroyed

Pool ocp-dr8v2-bootstrap has been undefined
```

**Comment by iranzo** [comment]
Check my conditionals on https://github.com/openshift-kni/baremetal-deploy/pull/597 so that code is valid for both lower and higher or equal than 4.6

**Comment by kurokobo** [comment]
Is the conditional for 4.5 or lower required for this repository?

The bare metal IPI is supported officially only from 4.6 or later, so the [documentation for 4.5](https://docs.openshift.com/container-platform/4.5/welcome/index.html) doesn't seem to have a link that leads to this page.

If it's required, of cource I'll add it.

**Comment by iranzo** [comment]
> Is the conditional for 4.5 or lower required for this repository?
> 
> The bare metal IPI is supported officially only from 4.6 or later, so the [documentation for 4.5](https://docs.openshift.com/container-platform/4.5/welcome/index.html) doesn't seem to have a link that leads to this page.
> 
> If it's required, of cource I'll add it.

Not for this one, but the changes we make in the upstream one (baremetal-deploy) will propagate to this repository so that they hit downstream documentation.

**Comment by kurokobo** [comment]
@iranzo
I really appreciate your support and I apologize to bother you by creating these PRs to this downstream repository first from outside of the team without a correct understanding of your internal process for review...

I added conditionals and for 4.7 too because openshift/installer#4448 seems to be cherry-picked into 4.7 only.

I'm glad if this will be approved, but if it's difficult to proceed, please feel free to close this. I'll keep my issue #27618 open for future tracking.

**Comment by iranzo** [comment]
> @iranzo
> I really appreciate your support and I apologize to bother you by creating these PRs to this downstream repository first from outside of the team without a correct understanding of your internal process for review...

No worries, I'm outside of the docs team as well, but as we try to keep both in sync, it's the approach we use to keep modules reusable as much as possible.

> 
> I added conditionals and for 4.7 too because [openshift/installer#4448](https://github.com/openshift/installer/pull/4448) seems to be cherry-picked into 4.7 only.
> 
> I'm glad if this will be approved, but if it's difficult to proceed, please feel free to close this. I'll keep my issue #27618 open for future tracking.

I'm not more than a contributor here,  let's see what @ahardin-rh has to say!

**Comment by ahardin-rh** [comment]
@kurokobo Thanks for this and sorry for the delay! This looks good to me and we appreciate your contributions! 
@iranzo @johnwilkins @rlopez133 Do you know who from QE can provide a review for this before we can merge? If you don't have a QE contact for this, I can help find the correct contact. Thanks so much!

**Comment by rlopez133** [comment]
Hi @ahardin-rh , I don't have a direct contact for this but likely someone in Amit Ugol's group. I'd reach out to him.

**Comment by ahardin-rh** [comment]
@rlopez133 Thanks! I reached out to him for a review and will stand by.

**Comment by tradej** [comment]
Tracked in TELCODOCS-218.

**Comment by ahardin-rh** [comment]
@johnwilkins FYI on this issue

**Comment by johnwilkins** [comment]
@ahardin-rh  I was under the impression that @rlopez133 had already fixed this issue.

**Comment by iranzo** [comment]
@johnwilkins @ahardin-rh yes, we've this fixed in the upstream docs, but seems we need to get them pulled into DS for the fix. The code we've in the upstream also have a conditional for latests 4.6 that uses a similar approach which was before only valid for > 4.7 and now it's > 4.6.

Code US is at: https://github.com/openshift-kni/baremetal-deploy/blob/master/documentation/ipi-install/modules/ipi-install-troubleshooting-cleaning-up-previous-installations.adoc#L18

**Comment by kurokobo** [comment]
Pushed additional fixes to match upstream.

Thank you very much for your support of this issue over a long period of time. I don't have access to TELCODOCS, so I have no way to know what's going on in detail. Hopefully I'd like to complete this but if you're having trouble or something difficult to proceed, please feel free to close this issue and start over with the standard flow of updating from upstream to downstream. Sorry for the inconvenience.

**Comment by tradej** [comment]
Thank you, @kurokobo. I have assigned @johnwilkins to this issue, he will be taking care of it.

**Comment by johnwilkins** [comment]
This PR can be closed. The fixes were made in https://github.com/openshift/openshift-docs/pull/31838

**Comment by kurokobo** [comment]
Thanks 😃

**Pull request closed without merging**